robot/client/client.go

```diff
@@ -806,6 +806,14 @@ func (rc *RobotClient) FrameSystemConfig(ctx context.Context) (*framesystem.Conf
 }
 
 // TransformPose will transform the pose of the requested poseInFrame to the desired frame in the robot's frame system.
+//
+//	  import (
+//		  "go.viam.com/rdk/referenceframe"
+//		  "go.viam.com/rdk/spatialmath"
+//	  )
+//
+//	  baseOrigin := referenceframe.NewPoseInFrame("test-base", spatialmath.NewZeroPose())
+//	  movementSensorToBase, err := robot.TransformPose(ctx, baseOrigin, "my-movement-sensor", nil)
```

Comment:
@raybjork does this seem like a reasonable/useful code example?

Reply:
Yeah I think this is fine

Reply:
My bad i forgot to include the link on the description but this example is taken directly from our exsiting Viam docs https://docs.viam.com/build/program/apis/robot/#transformpose. 

Once everything is in one place, I think we could make further improvements based on feedback or our own experience.